mdal/frmts/mdal_ugrid.cpp

```diff
@@ -695,65 +702,275 @@ void MDAL::DriverUgrid::save( const std::string &fileName, const std::string &,
   {
     MDAL::Log::error( err, name() );
   }
+
+  mNcFile.reset();
+}
+
+std::string MDAL::DriverUgrid::writeDatasetOnFileSuffix() const
+{
+  return "nc";
 }
 
 std::string MDAL::DriverUgrid::saveMeshOnFileSuffix() const
 {
   return "nc";
 }
 
-void MDAL::DriverUgrid::writeVariables( MDAL::Mesh *mesh )
+bool MDAL::DriverUgrid::writeDatasetGroup( MDAL::DatasetGroup *group, const std::string &fileName, const std::string &meshName )
 {
+  mNcFile.reset( new NetCDFFile );
+
+  try
+  {
+    mNcFile->openFile( fileName, true );
+  }
+  catch ( MDAL::Error &err )
+  {
+    err.setDriver( name() );
+    MDAL::Log::error( err );
+    return true;
+  }
```

Comment:
I would normally assume that you open file once in the persist function, add all groups/variables and then close a file. Why you need to open/close files for each dataset group? is it consistent with other drivers?

Reply:
Same logic for other drivers. Persisting dataset group concerns only one dataset group.

Reply:
ok cool